pkg/crypto/crypto.go

```diff
@@ -1,24 +1,24 @@
 package crypto
 
 import (
+	"bytes"
 	"crypto/sha256"
 	"fmt"
-	"io"
-	"os"
 
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/golang/protobuf/proto"
 	"github.com/juju/errors"
 )
 
-// SHA256 will return the sha256 sum of the provided filename content
-func SHA256(filename string) (string, error) {
-	hasher := sha256.New()
-	r, err := os.Open(filename)
-	if err != nil {
-		return "", errors.Trace(err)
-	}
-	defer r.Close()
-	if _, err := io.Copy(hasher, r); err != nil {
+// SHA256 will return the sha256 sum of the provided protocol buffer
+func SHA256(pb proto.Message) (string, error) {
+	m := jsonpb.Marshaler{Indent: "  "}
+	data := []byte{}
+	w := bytes.NewBuffer(data)
+
+	if err := m.Marshal(w, pb); err != nil {
 		return "", errors.Trace(err)
 	}
-	return fmt.Sprintf("%x", hasher.Sum(nil)), nil
+	sum := sha256.Sum256(w.Bytes())
+	return fmt.Sprintf("%x", sum), nil
```

Comment:
You don't need an intermediary buffer for this.
`sha256.New()` returns a [`hash.Hash`](https://golang.org/pkg/hash/#Hash), which implements the `io.Writer` interface. So this could be replaced by:

```suggestion
func SHA256(pb proto.Message) (string, error) {
    h := sha256.Hash() 
	err := jsonpb.Marshaler{Indent: "  "}.Marshal(h, pb)
	return fmt.Sprintf("%x", h.Sum(nil)), errors.Trace(err)
}


```

Reply:
nice!